src/cmds/qstat.c

```diff
@@ -1399,11 +1399,10 @@ display_statjob(struct batch_status *status, struct batch_status *prtheader, int
 							 */
 							char noval[] = "UNKNOWN";
 							prt_attr(a->name, a->resource, noval, alt_opt & ALT_DISPLAY_w);
-						} else
-						{
+						} else {
 							char time_buffer[32];
-							strcpy(time_buffer,ctime(&epoch));
-							time_buffer[strlen(time_buffer)-1]='\0';
+							pbs_strncpy(time_buffer, ctime(&epoch), sizeof(time_buffer));
+							time_buffer[strlen(time_buffer)-1] = '\0';
```

Comment:
ah, this line is needed because ctime returns a string with newline char at the end of the string. If we leave it then it causes bad formatting is qstat output.